corehq/apps/callcenter/tasks.py

```diff
@@ -46,6 +47,17 @@ def calculate_indicators():
         indicator_set.get_data()
 
 
+def sync_web_user_usercases_if_applicable(user, domain, spawn_task):
+    domain_obj = Domain.get_by_name(domain)
+    if settings.UNIT_TESTING:
+        return
```

Comment:
Somehow this just feels wrong to me. This makes this code and any calling code un-testable. I know there's not much code here to test and that the functions being called are tested elsewhere. Since this is already limited by a FF I think it would be sufficient to just check the FF:

```python
    if not USH_WEB_USER_CASE_CREATION.enabled(domain):
        return
    domain_obj = Domain.get_by_name(domain)
    if domain_obj.usercase_enabled:
        if spawn_task:
            sync_usercases_task.delay(user._id, domain)
        else:
            sync_usercases(user, domain)
```

Reply:
should I also change this in `sync_usercases_if_applicable`? I was just going by what is there.
```python
def sync_usercases_if_applicable(user, spawn_task):
    if settings.UNIT_TESTING and not user.project:
        return
    if (user.project.call_center_config.enabled or user.project.usercase_enabled):
        if spawn_task:
            sync_usercases_task.delay(user._id, user.domain)
        else:
            sync_usercases(user, user.domain)
```

Reply:
no I don't think there's a need to change that since there is no FF enabling that feature

Reply:
One thing to note, the `settings.UNIT_TESTING` check was added to prevent testing errors as a result of some `SimpleTestCases` that create a web user and then throw an error when the db accessor is called.

Reply:
as long as the FF is not enabled for those tests the it shouldn't be an issue

Reply:
I'm now realizing this was an issue before I even created the FF so I'm mistaken.

So would something like this work:
```python
def sync_web_user_usercases_if_applicable(user, domain):
    domain_obj = Domain.get_by_name(domain)
    if not domain_obj:
        return
    if domain_obj.usercase_enabled and USH_WEB_USER_CASE_CREATION.enabled(domain):
        sync_usercases_task.delay(user._id, domain)
```

Reply:
I think the code should fail if the domain is `None`. The other code combines this with a check on unit testing setting so that it will fail if the condition occurs in production. My suggestion for this function is slightly different since this has a FF so we shouldn't need the settings check. Any unit test that is executing this code with the FF enabled should also be creating the domain object in the DB.

Let me know if you want to go to voice on this.

Reply:
Okay that makes sense. Thanks for the clarification.
I think a quick 5 mins on voice would help tie a bow on this.